File: src/leetcode每日一题/_03_22/2038. 如果相邻两个颜色均相同则删除当前颜色.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int s = 0, n = colors.length();
        for(int i = 0, j = 0; i < n; i = j) {
            while(j < n && colors.charAt(j) == colors.charAt(i))
                j++;
            if(j - i >= 3)
                s += colors.charAt(i) == 'A' ? j - i - 2 : i + 2 - j;
        }
        return s > 0;
    }
}

File: src/leetcode每日一题/_03_22/2038. 如果相邻两个颜色均相同则删除当前颜色.go
package _03_21

func winnerOfGame(colors string) bool {
	s, n := 0, len(colors)
	for i, j := 0, 0; i < n; i = j {
		for j < n && colors[i] == colors[j] {
			j++
		}
		if v := j - i - 2; v > 0 {
			if colors[i] == 'A' {
				s += v
			} else {
				s -= v
			}
		}
	}
	return s > 0
}
